### PR TITLE
tests: introduce run_boot decorator for boot mode-specific tests

### PR DESCRIPTION
**Pull request by KKoukiou**
See also https://github.com/cockpit-project/bots/pull/7182

### CODE REVIEW

**Review by martinpitt — COMMENTED** [review]
Thanks! This looks good enough. Feels a bit weird to have both decorators and the Test..{EFI,Bios} class split -- feels like making the decorator work for classes would be a bit more elegant. But this is completely bikeshedding, and no need to waste time on that if you are happy with this.

[this failure](https://cockpit-logs.us-east-1.linodeobjects.com/pull-542-b48973ed-20241205-112656-fedora-rawhide-boot-other-efi/log) might be relevant or just a flake? I'll retry it to compare.

**Comment by KKoukiou** [comment]
> Thanks! This looks good enough. Feels a bit weird to have both decorators and the Test..{EFI,Bios} class split -- feels like making the decorator work for classes would be a bit more elegant. But this is completely bikeshedding, and no need to waste time on that if you are happy with this.
> 
> [this failure](https://cockpit-logs.us-east-1.linodeobjects.com/pull-542-b48973ed-20241205-112656-fedora-rawhide-boot-other-efi/log) might be relevant or just a flake? I'll retry it to compare.

Thhe Test..{EFI, BIOS} class split was eliminated in the second commit by the TEST_SCENARIO=efi :)

File: test/run [hunk 1]
```diff
@@ -21,6 +21,7 @@ RE_EXPENSIVE='E2E'
 
 # every known case needs to set RUN_OPTS to something non-empty, so that we can check if we hit any branch
 case "${TEST_SCENARIO:=}" in
+    *efi*) RUN_OPTS="$ALL_TESTS"; export TEST_FIRMWARE="efi" ;;&
```
Comment (rvykydal) [inline test/run:24]: I am confused by the '&' at the end of the line.

**Review by rvykydal — APPROVED** [review]
Looks good to me, thank you.

File: test/run (hunk 1 above)
Comment (martinpitt) [reply test/run:24]: It will make the `case` evaluation continue with the next one. I.e. the scenarios are not mutually exclusive, but combine.

Comment (KKoukiou) [reply test/run:24]: he ;& in a case statement in a Bash script allows execution to continue with the next pattern block in the case statement, regardless of whether the next pattern matches or not. 

The 'efi' test scenario it not mutually exclusive with the `expensive` and `other` scenarios, they can be used together.
So `expensive/efi`, `other/efi`. I will update the documentation in test/README.rst to reflect that.

**Pull request merged**